src/daft-logical-plan/src/ops/summarize.rs

```diff
@@ -15,7 +15,7 @@ pub fn summarize(input: &LogicalPlanBuilder) -> DaftResult<LogicalPlanBuilder> {
     let mut cnts: Vec<ExprRef> = vec![]; // count              :: int64
     let mut nuls: Vec<ExprRef> = vec![]; // nulls              :: int64
     let mut unqs: Vec<ExprRef> = vec![]; // approx_distinct    :: int64
-    for (_, field) in &input.schema().fields {
+    for field in input.schema().fields() {
```

Comment:
nit: if we had `impl Iter<Item=Field> for Schema` this might reduce all the `for f in input.schema().fields().iter()` or however it _exactly_ is so that we can have `for field in schema` like we have on the python side.

Reply:
That makes a lot of sense. Made the changes in this PR!